plugins/processors/dedup/dedup_test.go

```diff
@@ -152,3 +153,42 @@ func TestCacheShrink(t *testing.T) {
 
 	require.Equal(t, 0, len(deduplicate.Cache))
 }
+
+func TestSameTimestamp(t *testing.T) {
+	now := time.Now()
+	dedup := createDedup(now)
+	var in telegraf.Metric
+	var out []telegraf.Metric
+
+	in, _ = metric.New("metric",
+		map[string]string{"tag": "value"},
+		map[string]interface{}{"foo": 1}, // field
+		now,
+	)
+	out = dedup.Apply(in)
+	require.Equal(t, []telegraf.Metric{in}, out) // pass
```

Comment:
May want to use `testutil.RequireMetricsEqual` here for better error display.  (I'm not sure if this is testing the pointer or the value)